extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcRedirectFilter.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.oidc;
+
+import io.vertx.core.MultiMap;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * OIDC redirect filter which can be used to customize redirect requests to OIDC authorization and logout endpoints
+ * as well as local redirects to OIDC tenant error, session expired and other pages.
+ */
+public interface OidcRedirectFilter {
+
+    /**
+     * OIDC redirect context which provides access to the routing context, current OIDC tenant configurationthe redirect uri
+     * which filters may update by adding additional query parameters,
+     * redirect URI and additional query parameters.
+     */
+    interface OidcRedirectContext {
```

Comment:
This can also be a record if you remove the `get` prefix from the getters ;)

Reply:
Let's do the modern API :-)

Reply:
@gastaldi I'm feeling like a good Java engineer again :-)